stdlib/src/posixsubprocess.rs

```diff
@@ -121,21 +120,95 @@ impl CharPtrSlice<'_> {
     }
 }
 
+#[derive(Copy, Clone)]
+struct Fd(BorrowedFd<'static>);
+
+impl TryFromObject for Fd {
+    fn try_from_object(vm: &VirtualMachine, obj: PyObjectRef) -> PyResult<Self> {
+        match MaybeFd::try_from_object(vm, obj)? {
+            MaybeFd::Some(fd) => Ok(fd),
+            MaybeFd::None => Err(vm.new_value_error("invalid fd".to_owned())),
+        }
+    }
+}
+
+impl Write for Fd {
+    fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
+        Ok(unistd::write(self, buf)?)
+    }
+
+    fn flush(&mut self) -> std::io::Result<()> {
+        Ok(())
+    }
+}
+
+impl AsRawFd for Fd {
+    fn as_raw_fd(&self) -> RawFd {
+        self.0.as_raw_fd()
+    }
+}
+
+impl IntoRawFd for Fd {
+    fn into_raw_fd(self) -> RawFd {
+        self.0.as_raw_fd()
+    }
+}
+
+impl AsFd for Fd {
+    fn as_fd(&self) -> BorrowedFd<'_> {
+        self.0.as_fd()
+    }
+}
+
+impl From<OwnedFd> for Fd {
+    fn from(fd: OwnedFd) -> Self {
+        Self(unsafe { BorrowedFd::borrow_raw(fd.into_raw_fd()) })
+    }
+}
+
+#[derive(Copy, Clone)]
+enum MaybeFd {
```

Comment:
Wouldn't it be more correct to have:
```rust
type MaybeFd = Option<(FD type)>
```

or maybe even a newtype:
```rust
struct MaybeFd<T>(Option<T>)
```
?

Reply:
In this case, None corresponds to `-1`, not `PyNone`

Reply:
ah, I see. Then this is probably the best I can think about atm

Reply:
Is Some/None naming the best? It looks a bit confusing. 

If it needs option-like operations, `#[derive(result_like::OptionLike)]` will be helpful.

Reply:
Good point. I've renamed the variants, which I think makes it more clear.